types/coin.go

```diff
@@ -485,6 +487,15 @@ var (
 	reDecCoin = regexp.MustCompile(fmt.Sprintf(`^(%s)%s(%s)$`, reDecAmt, reSpc, reDnm))
 )
 
+func validateDenom(denom string) {
+	if len(denom) < 3 || len(denom) > 16 {
```

Comment:
look like magic numbers to me

Reply:
Hmm how so? This is just for validation, we can pick any consistent length range.